src/node_data.cpp

```diff
@@ -256,9 +256,21 @@ void node_data::reset_map() {
 }
 
 void node_data::insert_map_pair(node& key, node& value) {
-  m_map[&key] = &value;
+
+  bool found = false;
```

Comment:
I see your concern, and share it...
Adding std::unordered_map<node*, size_t> is simple and will take care of it
;)